File: strings/roman_to_integer.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    char roman [] = {'I','V','X','L','C','D','M'};
    int roman_int [] = {1,5,10,50,100,500,1000};

    int result = 0;
    string s ="MCMXCIV";
    for(int i = 0; i < s.length();i++)
    {
        for(int j=0;j< strlen(roman);j++){
                if(s[i]==roman[j]){
                    result+=roman_int[j];
                }
                

        }
    }
    cout<<result;
}

File: strings/roman_to_integer.js
const print = console.log;
var romanToInt = function (s) {
  const roman = new Map([
    ['I', 1],
    ['V', 5],
    ['X', 10],
    ['L', 50],
    ['C', 100],
    ['D', 500],
    ['M', 1000],
  ]);
  let result = 0;
  for (let i = 0; i < s.length; i++) {
    if (i < s.length && roman.get(s[i]) < roman.get(s[i + 1])) {
      result -= roman.get(s[i]);
    } else result += roman.get(s[i]);
  }

  
  return result;
};

print(romanToInt('MCMXCIV'));
